index.js

```diff
@@ -1,15 +1,58 @@
 'use strict';
+const Executor = require('screwdriver-executor-base');
+const fs = require('fs');
+const request = require('request');
+const tinytim = require('tinytim');
+const yaml = require('js-yaml');
+const API_KEY = fs.readFileSync('/etc/kubernetes/apikey', 'utf-8');
+const SCM_URL_REGEX = /^git@([^:]+):([^\/]+)\/(.+?)\.git(#.+)?$/;
+const GIT_ORG = 2;
+const GIT_REPO = 3;
+const GIT_BRANCH = 4;
 
-const Executor = class {
+const jobsUrl = 'https://kubernetes/apis/batch/v1/namespaces/default/jobs';
+
+class K8sExecutor extends Executor {
     /**
      * Starts a k8s build
      * @method start
-     * @param  {Object}   config   A configuration object
+     * @param  {Object}   config            A configuration object
+     * @param  {Object}   config.buildId    ID for the build
+     * @param  {Object}   config.jobId      ID for the job
+     * @param  {Object}   config.pipelineId ID for the pipeline
+     * @param  {Object}   config.container  Container for the build to run in
+     * @param  {Object}   config.scmUrl     Scm URL to use in the build
      * @param  {Function} callback Callback function
      */
     start(config, callback) {
-        process.nextTick(callback);
+        const scmMatch = config.scmUrl.match(SCM_URL_REGEX);
```

Comment:
Reverse this.  If `config.scmUrl` doesn't exist or isn't a string, this will fail in weird ways.

You want `SCM_URL_REGEX.exec(config.scmUrl)`